motionplan/armplanning/planner_options.go

```diff
@@ -213,6 +223,46 @@ func (p *PlannerOptions) getGoalMetric(goal referenceframe.FrameSystemPoses) mot
 	}
 }
 
+func (p *PlannerOptions) getGoalMetricLinear(goal referenceframe.FrameSystemPoses) (motionplan.LinearFSMetric, error) {
+	if p.GoalMetricType != motionplan.SquaredNormOpt {
+		//nolint
+		return nil, fmt.Errorf("May only call `getGoalMetricLinear` with a planner type of `SquaredNormOpt`")
+	}
+
+	metrics := map[string]motionplan.StateMetric{}
+	frames := map[string]string{}
+
+	for frame, goalInFrame := range goal {
+		frames[frame] = goalInFrame.Parent()
+		metrics[frame] = motionplan.NewSquaredNormMetric(goalInFrame.Pose())
+	}
+
+	return func(state *motionplan.LinearFS) float64 {
+		score := 0.
+		for frame, goalMetric := range metrics {
+			poseParent := frames[frame]
+			var currPose spatialmath.Pose
+
+			if p.GoalMetricType == motionplan.SquaredNormOpt {
+				dq, err := state.FS.TransformOptLinear(state.Configuration, frame, poseParent)
+				if err != nil {
+					panic(fmt.Sprintf("fs: %v frame: %s poseParent: %v err: %v",
```

Comment:
how is this condition possible? only if there is a programmer error in RDK?

why panic here & error here: https://github.com/viamrobotics/rdk/pull/5422/files#diff-4660f93c6d15c86f7749692e5164d0e0a702f6a2a7657f82b366654603ddb3f4R229

Reply:
It's just copied over. Happy to tidy up after there's once the final "big strokes" are agreed on.